candid/src/de.rs

```diff
@@ -777,3 +777,57 @@ impl<'de, 'a> de::VariantAccess<'de> for Compound<'a, 'de> {
         }
     }
 }
+
+pub trait DecodeArguments<'a>: Sized {
+    fn decode_arguments(_de: IDLDeserialize<'a>) -> Result<(IDLDeserialize<'a>, Self)>;
+}
+
+// // Is this a sensible impl?
+impl<'a> DecodeArguments<'a> for () {
```

Comment:
zero argument is a tricky case. We want to distinguish between zero argument and one argument with `()` as a value.

Reply:
So one argument with a value of void is `((),)`.

Reply:
Yeah, `(())` vs `((),)`. At least worthing documenting in the comment.